lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1606,6 +1627,76 @@ void ReduceOpConversion::accumulate(ConversionPatternRewriter &rewriter,
   case RedOp::XOR:
     acc = xor_(acc, cur);
     break;
+  case RedOp::ARGMIN:
+  case RedOp::ARGMAX:
+  case RedOp::ARGUMIN:
+  case RedOp::ARGUMAX:
+  case RedOp::ARGFMIN:
+  case RedOp::ARGFMAX:
+    llvm::report_fatal_error(
+        "This accumulate implementation is not for argmin / argmax");
+  default:
+    llvm::report_fatal_error("Unsupported reduce op");
+  }
+}
+
+void ReduceOpConversion::accumulate(ConversionPatternRewriter &rewriter,
+                                    Location loc, RedOp redOp, Value &acc,
+                                    Value &accIndex, Value cur, Value curIndex,
+                                    bool isFirst) const {
+  if (isFirst) {
+    acc = cur;
+    accIndex = curIndex;
+    return;
+  }
+  switch (redOp) {
```

Comment:
nit: the switch/case could be much simpler with templates

Reply:
Can you please give a very short description of the template structure? Thanks 😃

Reply:
Pls ignore, i'm just aware about the difference in select(..). I think it's fine to keep it as it is.